### ISSUE TITLE
multi table

### ISSUE THREAD

**Issue by zeuz10** [diprokon/ng-table-virtual-scroll#93]
doesn't seem to be working I have the following code:

ts file<<<<<<<<<
    this.tables = [];
    this.tables.push(new TableVirtualScrollDataSource(this.dataSourceDetalle.filter(f => f.WERKS == tienda)
      .sort((a,b) => (a.SBDEPTO < b.SBDEPTO) ? 1 : ((b.SBDEPTO < a.SBDEPTO) ? -1 : 0)))
);

HTML file <<<<<<<<<
 <cdk-virtual-scroll-viewport tvsItemSize
                               class="wrapper mat-elevation-z2">
  <table mat-table [dataSource]="table" *ngFor="let table of tables" matSort [class]="status=='d' ? 'light-blue':''">
.
.
. having about a matrix for about 300 rows and 200+ columns spread along 3 generated tables

**Comment by diprokon** [comment]
Not sure how to help you
Did you import all required modules?
Can you provide stackblitz example?

**Comment by zeuz10** [comment]
Yes, all modules are in place, the table was working before the virtual scroll.
I don't have the stackblitz example, let me work on that

**Comment by zeuz10** [comment]
After a while of debugging, I have the following
Headers, footers
main issue was that for the data to get the footers I used the table object and not table.data or table.filteredData
 but no content. instead, I get this error

ERROR TypeError: Cannot read properties of undefined (reading '_switchDataSource')
    at TableItemSizeDirective.ngAfterContentInit (ng-table-virtual-scroll.mjs:187:40)

I believe is some kind of configuration, data comes from the server in an http.get request from HttpClient so ngAfterContentInit is empty until data arrives.
the thing is... I get all the calculated data from the table in the footers
